Detections/AuditLogs/AdminPromoAfterRoleMgmtAppPermissionGrant.yaml

```diff
@@ -0,0 +1,74 @@
+id: f80d951a-eddc-4171-b9d0-d616bb83efdc
+name: Admin promotion after Role Management Application Permission Grant
+description: |
+  'This rule looks for a service principal being granted the Microsoft Graph RoleManagement.ReadWrite.Directory (application) permission before being used to add an Azure AD object or user account to an Admin directory role (i.e. Global Administrators). 
+  Ref : https://docs.microsoft.com/en-us/graph/permissions-reference#role-management-permissions, https://docs.microsoft.com/en-us/graph/api/directoryrole-post-members?view=graph-rest-1.0&tabs=http'
+severity: High
+requiredDataConnectors:
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AuditLogs
+queryFrequency: 2h
+queryPeriod: 2h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+  - Persistence
+relevantTechniques:
+  - T1098.003
+tags:
+  - SimuLand
+query: |
+  AuditLogs
+  | where LoggedByService =~ "Core Directory"
+  | where Category =~ "ApplicationManagement"
+  | where AADOperationType =~ "Assign"
+  | where ActivityDisplayName =~ "Add app role assignment to service principal"
+  | mv-expand TargetResources
+  | mv-expand TargetResources.modifiedProperties
+  | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
+  | where displayName_ =~ "AppRole.Value"
+  | extend AppRole = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
+  | where AppRole contains "RoleManagement.ReadWrite.Directory"
+  | extend InitiatingApp = tostring(parse_json(tostring(InitiatedBy.app)).displayName)
+  | extend Initiator = iif(isnotempty(InitiatingApp), InitiatingApp, tostring(parse_json(tostring(InitiatedBy.user)).userPrincipalName))
+  | extend Target = tostring(parse_json(tostring(TargetResources.modifiedProperties[4].newValue)))
+  | extend TargetId = tostring(parse_json(tostring(TargetResources.modifiedProperties[3].newValue)))
+  | summarize by bin(TimeGenerated, 1h), OperationName, Initiator, Target, TargetId, Result
+  | join (
+    AuditLogs
+    | where LoggedByService =~ "Core Directory"
+    | where Category =~ "RoleManagement"
+    | where AADOperationType in ("Assign", "AssignEligibleRole")
+    | where ActivityDisplayName has_any ("Add eligible member to role", "Add member to role")
+    | mv-expand TargetResources
+    | mv-expand TargetResources.modifiedProperties
+    | extend displayName_ = tostring(TargetResources_modifiedProperties.displayName)
+    | where displayName_ =~ "Role.DisplayName"
+    | extend RoleName = tostring(parse_json(tostring(TargetResources_modifiedProperties.newValue)))
+    | where RoleName contains "Admin"
```

Comment:
You should be able to do the same here with "has" versus "contains"

Reply:
This will not work in here because `has` looks for exact string values. With `contains` we want to capture `X Admin Group` and `Global Administrators`. I tested it and `has` did not show any results. However, `contains` did.

Reply:
Got it, so because "Admin" is part of "Administrator" then 'has' misses it.